src/Adapter/Image/Uploader/TitleImageUploader.php

```diff
@@ -0,0 +1,115 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\Image\Uploader;
+
+use Gender;
+use ImageManager;
+use PrestaShop\PrestaShop\Core\Domain\Title\Exception\TitleImageUploadingException;
+use PrestaShop\PrestaShop\Core\Image\Uploader\Exception\ImageUploadException;
+use PrestaShop\PrestaShop\Core\Image\Uploader\Exception\UploadedImageConstraintException;
+use PrestaShop\PrestaShop\Core\Image\Uploader\ImageUploaderInterface;
+use Symfony\Component\HttpFoundation\File\UploadedFile;
+
+/**
+ * Uploads title logo image
+ */
+class TitleImageUploader extends AbstractImageUploader implements ImageUploaderInterface
+{
+    /**
+     * @var int|null
+     */
+    protected $imageHeight;
+    /**
+     * @var int|null
+     */
+    protected $imageWidth;
+
+    /**
+     * {@inheritdoc}
+     *
+     * @throws UploadedImageConstraintException
+     * @throws ImageUploadException
+     */
+    public function upload($titleId, UploadedFile $image)
```

Comment:
```suggestion
    public function upload($titleId, UploadedFile $image, int $imageWidth, int $imageHeight)
```
Why not specify the sizes as parameters, this would prevent making this service stateful, especially since all the usages so far always call the two setters, then the upload method. So maybe only `upload` could be enough as a public method.

Reply:
Oh I think I get it! That's because of the `ImageUploaderInterface` right?
I believe that you could add two optional parameters without breaking the interface, that's actually been done a few times with implementations of `UrlProviderInterface` that needed some parameters.

That said it becomes not so much interesting to use an interface in this case if it doesn't meet the need, which kind of make me wonder if you should implement this interface here, even `AbstractTitleHandler` doesn't rely on the interface but the implementation itself. Maybe extending `AbstractImageUploader` is enough?